src/promptflow/promptflow/contracts/flow.py

```diff
@@ -853,3 +853,22 @@ def _replace_with_variant(self, variant_node: Node, variant_tools: list):
                 self.nodes[index] = variant_node
                 break
         self.tools = self.tools + variant_tools
+
+
+@dataclass
+class EagerFlow:
+    # region: data plane flow definition
+    inputs: Dict[str, InputDefinition]
```

Comment:
Not sure if we should use tool's input definition or flow input's definition.
Maybe we can define eager flow's own input.
Currently we don't have connections in inputs and don't have chat inputs.

Reply:
why we should use a definition different from standard flow?

Reply:
Basically because

- No chat inputs/outputs for eager flow.
- eager flow outputs don't have reference.